pkg/cmd/codespace/common.go

```diff
@@ -60,8 +61,18 @@ func (a *App) StopProgressIndicator() {
 	a.io.StopProgressIndicator()
 }
 
+type liveshareSession interface {
+	Close() error
+	GetSharedServers(context.Context) ([]*liveshare.Port, error)
+	KeepAlive(string)
+	OpenStreamingChannel(context.Context, liveshare.ChannelID) (ssh.Channel, error)
+	StartJupyterServer(context.Context) (int, string, error)
+	StartSharing(context.Context, string, int) (liveshare.ChannelID, error)
+	StartSSHServer(context.Context) (int, string, error)
```

Comment:
Thanks for the suggestion @mislav  -- this is a much smaller interface to work with 👍🏻